### PR TITLE
Add the ability to create global pointer routes

### PR DESCRIPTION
**Pull request by abarth**
We'll use this functionality to implement some subtle behaviors for tooltips.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter/lib/src/gestures/pointer_router.dart [hunk 1]
```diff
@@ -38,35 +39,60 @@ class PointerRouter {
...
+        exception: exception,
+        stack: stack,
+        library: 'gesture library',
+        context: 'while routing a pointer event',
+        router: this,
+        route: route,
+        event: event,
+        informationCollector: (StringBuffer information) {
+          information.writeln('Event:');
+          information.write('  $event');
```
Comment (Hixie) [inline packages/flutter/lib/src/gestures/pointer_router.dart]: should include information about whether it was a global route or not

File: packages/flutter/test/gestures/pointer_router_test.dart [hunk 2]
```diff
@@ -41,4 +41,56 @@ void main() {
...
+      router.addGlobalRoute(secondCallback);
+    });
+
+    TestPointer pointer2 = new TestPointer(2);
+    router.route(pointer2.down(Point.origin));
+    expect(firstCallbackRan, isTrue);
+    expect(secondCallbackRan, isFalse);
+  });
+
+  test('Supports re-entrant global cancellation', () {
```
Comment (Hixie) [inline packages/flutter/test/gestures/pointer_router_test.dart]: the docs should talk about this :-)

File: packages/flutter/test/gestures/pointer_router_test.dart [hunk 3]
```diff
@@ -41,4 +41,56 @@ void main() {
...
+    router.addRoute(2, (PointerEvent event) {
+      perPointerCallbackRan = true;
+      router.addGlobalRoute(callback);
+    });
+    TestPointer pointer2 = new TestPointer(2);
+    router.route(pointer2.down(Point.origin));
+    expect(perPointerCallbackRan, isTrue);
+    expect(callbackRan, isFalse);
+  });
+
```
Comment (Hixie) [inline packages/flutter/test/gestures/pointer_router_test.dart]: one final thing to check is the relative order of pointer and global routes

**Comment by Hixie** [comment]
LGTM

**Pull request merged**